pkg/logs/docker/docker.go

```diff
@@ -41,5 +44,56 @@ func ParseMessage(msg []byte) (string, string, []byte, error) {
 	}
 	to += messageHeaderLength
 	ts := string(msg[messageHeaderLength:to])
+
 	return ts, status, msg[to+1:], nil
+
+}
+
+// getPreMessageLength finds length of the 8 byte header, timestamp, and space
+// the is in front of each 16Kb chunk of message
+func getPreMessageLength(msg []byte) int {
+	idx := bytes.Index(msg, []byte{' '})
+	if idx == -1 {
+		return 0
+	}
+	return idx + 1
+}
+
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+// removePartialHeaders removes the 8 byte header, timestamp, and space
+// that occurs between each 16Kb section of a log greater than 16 Kb in length.
+// If a docker log is greater than 16Kb, each 16Kb section "PartialMessage" will
+// have a header, timestamp, and space in front of it.  For illustration,
+// let's call this "HeadTs ".  For example, a message that is 35kb will be of the form:
+// `HeadTs PartialMessageHeadTs PartialMessageHeadTs PartialMessage`
+// This function removes the "HeadTs " between two PartialMessage sections while
+// leaving the very first "HeadTs "
+func removePartialHeaders(msg []byte) []byte {
+	preMessageLength := getPreMessageLength(msg)
+	if len(msg[preMessageLength:]) < maxDockerBufferSize {
+		return msg
+	}
+
+	removed := msg[:preMessageLength+maxDockerBufferSize]
+	msg = msg[preMessageLength+maxDockerBufferSize:]
+	preMessageLength = getPreMessageLength(msg)
+	nextPartialMessageSize := min(len(msg), maxDockerBufferSize+preMessageLength)
+
+	// When there is more log message left to handle, loop through what remains,
+	// and build the message while removing the header, timestamp, and space that
+	// separates each 16Kb of log
+	for nextPartialMessageSize > 0 {
+		removed = append(removed, msg[preMessageLength:nextPartialMessageSize]...)
```

Comment:
`removed`: I think we could find a better name, `removed = append` isn't very clear